### PR TITLE
EVA-2033 Refactor eva accession

### PR DESCRIPTION
**Pull request by andresfsilva**
Merge EBIVariation/variation-commons#96 before this PR

- [x] Implement getAllByAccession to return a List
- [x] Remove findByHashedMessageIn
- [x] Implement getByIdentifyingFields
- [x] Implement getLastInactive 
- [x] Use getVariantClassification without subSnpClass from variation-commons

### CODE REVIEW

File: eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/service/nonhuman/ClusteredVariantAccessioningService.java [hunk 1]
```diff
@@ -210,12 +213,11 @@ public void merge(Long accessionOrigin, Long mergeInto, String reason)
     }
 
     public AccessionWrapper<IClusteredVariant, String, Long> getLastInactive(Long accession) {
-        throw new UnsupportedOperationException("TODO: This should be implemented, but I forgot to.");
```
Comment (jmmut) [inline eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/service/nonhuman/ClusteredVariantAccessioningService.java:213]: I knew I would forget this XD

File: eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/service/nonhuman/dbsnp/DbsnpClusteredVariantMonotonicAccessioningService.java [hunk 2]
```diff
@@ -53,4 +55,13 @@ public String getHash(IClusteredVariant variant) {
     public List<AccessionWrapper<IClusteredVariant, String, Long>> getByHash(List<String> hashes) {
```
Comment (jmmut) [inline eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/service/nonhuman/dbsnp/DbsnpClusteredVariantMonotonicAccessioningService.java:55]: can't you use the function in the parent? you deleted this function from the Submitted classes.

File: eva-accession-ws/src/main/java/uk/ac/ebi/eva/accession/ws/rest/ClusteredVariantsRestController.java [hunk 3]
```diff
@@ -71,25 +67,16 @@
 
     private ClusteredVariantAccessioningService nonHumanActiveService;
 
-    // TODO don't use the dbsnpInactiveService. This won't return EVA accessioned ClusteredVariants. A method
```
Comment (jmmut) [inline eva-accession-ws/src/main/java/uk/ac/ebi/eva/accession/ws/rest/ClusteredVariantsRestController.java:74]: feels so good to resolve TODOs

File: eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/service/nonhuman/dbsnp/DbsnpClusteredVariantMonotonicAccessioningService.java (hunk 2 above)
Comment (andresfsilva) [reply eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/service/nonhuman/dbsnp/DbsnpClusteredVariantMonotonicAccessioningService.java]: Thanks, I missed that

**Pull request merged**